pkg/build/builder/sti.go

```diff
@@ -451,7 +451,9 @@ func (s *S2IBuilder) pullImage(name string, authConfig dockerclient.AuthConfigur
 		options.Repository = name
 	}
 
-	return s.dockerClient.PullImage(options, authConfig)
+	return retryImageAction("Pull", func() (pullErr error) {
+		return s.dockerClient.PullImage(options, authConfig)
+	})
```

Comment:
don't we also do this in docker builder?

Reply:
Yes, this is what DockerBuilder does.

Reply:
also extractSourceFromImage in source.go pulls images indirectly (it calls Mount which i assume pulls the image) so we might need to retry the Mount if it fails due to a pull failure also.

Reply:
oh I see, it was only missing in the s2i path.  got it.

Still wondering if we need to do something in the Mount flow when we're extracting content from other images?

Reply:
Oh, I see what you mean now.  `extractSourceFromImage()` depends on `buildah.NewBuilder()`  to implicitly pull the image that we want to extract content from, so it wasn't as obvious that we weren't retrying there.